src/ckb-daemon/usb_mac.c

```diff
@@ -271,6 +271,67 @@ static void pipecomplete(void* refcon, IOReturn result, void* arg0){
     (*handle)->ReadPipeAsync(handle, ctx->pipe, buffer, ctx->maxsize, pipecomplete, ctx);
 }
 
+CFMachPortRef event_tap;
+
+CGEventRef cg_callback(CGEventTapProxy proxy, CGEventType type, CGEventRef event, void* refcon) {
+    if (type == kCGEventTapDisabledByTimeout) {
+        CGEventTapEnable(event_tap, true);
+        return event;
+    }
+
+    if (event) {
+        for(int i = 0; i < DEV_MAX; i++){
+            if(IS_CONNECTED(keyboard + i)){
+                usbdevice *kb = keyboard + i;
+                if (!IS_MOUSE_DEV(kb)) {
+                    pthread_mutex_lock(imutex(kb));
+                    CGEventSetFlags(event, (kCGEventFlagMaskNonCoalesced | kb->modifiers));
+                    pthread_mutex_unlock(imutex(kb));
+                    break;
```

Comment:
It seems to me, that there is the following assumption: There is at most 1 non-mouse keyboard device. Otherwise I don't understand the `break` statement after there are certain flags set for the given event.

Especially here some comments might come in handy.